sdk/core/azure_core/src/http/pager.rs

```diff
@@ -145,125 +417,147 @@ impl<T> futures::Stream for PageStream<T> {
     }
 }
 
-impl<T> std::fmt::Debug for PageStream<T> {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("Pager").finish_non_exhaustive()
+impl<P> fmt::Debug for PageIterator<P> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("PageIterator").finish_non_exhaustive()
     }
 }
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 enum State<T> {
     Init,
-    Continuation(T),
+    More(T),
     Done,
 }
 
-#[cfg(test)]
-mod tests {
-    use std::collections::HashMap;
+fn iter_from_callback<
```

Comment:
non-blocking nit: This could return `impl Stream` and you could use `ItemIterator::from_stream` to reduce duplication a bit here. No functional difference there though, so I'm fine either way.